keras_hub/src/models/roformer_v2/roformer_v2_backbone.py

```diff
@@ -0,0 +1,214 @@
+from keras import activations
+from keras import ops
+
+from keras_hub.src.api_export import keras_hub_export
+from keras_hub.src.layers.modeling.reversible_embedding import (
+    ReversibleEmbedding,
+)
+from keras_hub.src.models.backbone import Backbone
+from keras_hub.src.models.roformer_v2.roformer_v2_attention import RoformerNorm
+from keras_hub.src.models.roformer_v2.roformer_v2_encoder import (
+    RoformerV2Encoder,
+)
+
+
+def roformer_kernel_initializer(stddev=0.02):
+    return keras.initializers.TruncatedNormal(stddev=stddev)
+
+
+@keras_hub_export("keras_hub.models.RorformerV2Backbone")
+class RoformerV2Backbone(Backbone):
+    """A RoformerV2 encoder network.
+
+    This class implements a bi-directional Transformer-based encoder as
+    described in ["Roformer"](https://github.com/ZhuiyiTechnology/roformer).
+    It includes the
+    embedding lookups and transformer layers, but not the masked language model
+    or next sentence prediction heads.
+
+    The default constructor gives a fully customizable, randomly initialized
+    RoformerV2 encoder with any number of layers, heads, and embed dim.To
+    load preset architectures and weights, use the `from_preset()` constructor.
+
+    Disclaimer: Pre-trained models are provided on an "as is" basis, without
+    warranties or conditions of any kind.
+
+    Args:
+        vocabulary_size: int. The size of the token vocabulary.
+        num_layers: int. The number of transformer layers.
+        num_heads: int. The number of attention heads for each transformer.
+            The hidden size must be divisible by the number of attention heads.
+        hidden_dim: int. The size of the transformer encoding and pooler layers.
+        intermediate_dim: int. The output dimension of the first Dense layer in
+            a two-layer feedforward network for each transformer.
+        dropout: float. Dropout probability for the Transformer encoder.
+        num_segments: int. The number of types that the 'segment_ids' input can
+            take.
+        dtype: string or `keras.mixed_precision.DTypePolicy`. The dtype to use
+            for model computations and weights. Note that some computations,
+            such as softmax and layer normalization, will always be done at
+            float32 precision regardless of dtype.
+
+    Examples:
+    ```python
+    input_data = {
+        "token_ids": np.ones(shape=(1, 12), dtype="int32"),
+        "segment_ids": np.array([[0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0]]),
+        "padding_mask": np.array([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]]),
+    }
+
+    # Pretrained RoformerV2 encoder.
+    model = keras_hub.models.RoformerV2Backbone.from_preset("roformerV2_base")
+    model(input_data)
+
+    # Randomly initialized RoformerV2 encoder with a custom config.
+    model = keras_hub.models.RoformerV2Backbone(
+        vocabulary_size=30552,
+        num_layers=4,
+        num_heads=4,
+        hidden_dim=256,
+        intermediate_dim=512,
```

Comment:
I don't think this runs, given that `head_size` is a required arg and not present here. Try running this. (the from_preset one won't run till we uplaod, that's expected)